packages/core-blockchain/lib/blockchain.js

```diff
@@ -263,80 +264,81 @@ module.exports = class Blockchain {
 
         state.lastBlock = block
 
-        callback()
+        return callback()
       } else if (block.data.height > this.getLastBlock().data.height + 1) {
         state.lastDownloadedBlock = state.lastBlock
-        callback()
+        return callback()
       } else if (block.data.height < this.getLastBlock().data.height || (block.data.height === this.getLastBlock().data.height && block.data.id === this.getLastBlock().data.id)) {
         state.lastDownloadedBlock = state.lastBlock
-        callback()
+        return callback()
       } else {
         state.lastDownloadedBlock = state.lastBlock
         logger.info(`Block ${block.data.height.toLocaleString()} disregarded because on a fork :knife_fork_plate:`)
-        callback()
+        return callback()
       }
     } else {
       logger.warn(`Block ${block.data.height.toLocaleString()} disregarded because verification failed :scroll:`)
-      callback()
+      return callback()
     }
   }
 
   /**
    * Process the given block.
+   * We should be sure this is fail safe (ie callback() is being called)
```

Comment:
This is a `TODO:`.

Reply:
no i have made it failsafe, putting it as a reminder if somebody touch it

Reply:
Then `NOTE:` to make it clearly visible that it should be read.

Reply:
OK